cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -184,7 +185,42 @@ func generateMountsModeRecover() error {
 }
 
 func generateMountsModeRun() error {
-	return fmt.Errorf("run mode mount generation not implemented yet")
+	seedDir := filepath.Join(runMnt, "ubuntu-seed")
+	bootDir := filepath.Join(runMnt, "ubuntu-boot")
+	dataDir := filepath.Join(runMnt, "ubuntu-data")
+
+	// 1. always ensure basic partitions are mounted
+	for _, d := range []string{seedDir, bootDir, dataDir} {
+		isMounted, err := osutilIsMounted(d)
+		if err != nil {
+			return err
+		}
+		if !isMounted {
+			fmt.Fprintf(stdout, "/dev/disk/by-label/%s %s\n", filepath.Base(d), d)
+			return nil
+		}
+	}
+	// 2. read modeenv
+	modeEnv, err := boot.ReadModeenv(filepath.Join(dataDir, "system-data"))
+	if err != nil {
+		return err
+	}
+	// 3. mount base
+	isMounted, err := osutilIsMounted(filepath.Join(runMnt, "base"))
```

Comment:
It seems that `err` is not really being used here?

Reply:
Yes, silly me. Fixed.